tools/clang/lib/Parse/ParseDecl.cpp

```diff
@@ -4135,7 +4135,7 @@ void Parser::ParseDeclarationSpecifiers(DeclSpec &DS,
     case tok::kw_union: {
       // HLSL Change Starts
       if (getLangOpts().HLSL) {
-        if (Tok.is(tok::kw_union) || Tok.is(tok::kw___interface)) {
+        if (Tok.is(tok::kw_union) || Tok.is(tok::kw___interface) || Tok.is(tok::kw_interface)) {
```

Comment:
I've made changes you suggested, thanks. There was one assert I removed that seems unecessary at this point. I didn't understand if I need to revert any rewriter files.

Reply:
Ideally you should be able to revert the rewriter test cases because the rewriter should still properly handle the HLSL 2015 `interface` keyword.

Reply:
> Ideally you should be able to revert the rewriter test cases because the rewriter should still properly handle the HLSL 2015 `interface` keyword.

Reverting the rewriter files (cpp-errors_noerr.hlsl, cpp-errors_gold.hlsl) causes "error: 'interface' is a reserved keyword in HLSL" in cpp-errors_noerr.hlsl.

Reply:
Oh, you may need to add `-HV 2015` to the RUN line. The rewriters need to work with 2015 inputs, but they do some odd things.

Reply:
Strange but even with adding `-HV 2015`  it still shows the same error. I used the same RUN line as cpp-errors-hv2015.hlsl which passes just fine:
// RUN: %clang_cc1 -HV 2015 -fsyntax-only -Wno-unused-value -ffreestanding -verify %s

Reply:
Oh. Even cooler... The rewriter tests don't read the run line at all, they just use hard coded arguments.

Reply:
Oh, in that case do I change rewriter's version to 2015 then or leave it as it is?